### PR TITLE
KeyCloak integration for secure-embedded-cache quickstart

### PR DESCRIPTION
**Pull request by mgencur**
This PR replaces https://github.com/jboss-developer/jboss-jdg-quickstarts/pull/205

### CODE REVIEW

File: secure-embedded-cache/README.md [hunk 1]
```diff
@@ -33,53 +35,69 @@ Setup
...
+4. Download Red Hat Single Sign-On (RH-SSO) server and Client Adapter for JBoss EAP 7 from Red Hat's Customer portal.
+
+5. Install the Client Adapter in EAP by unpacking the Client Adapter zip file into the root directory of EAP
+   and running the following command:
+
+   ${JBOSS_HOME}/jboss-cli.sh --file=adapter-install-offline.cli
+
+6. Start RH-SSO/Keycloak server with port offset:
+
+    ${KEYCLOAK_HOME}/bin/standalone.sh -Djboss.socket.binding.port-offset=100
```
Comment (slaskawi) [inline secure-embedded-cache/README.md:52]: I think it might be easier to use Keycloak docker image.

Comment (slaskawi) [reply secure-embedded-cache/README.md:52]: Alternatively, Keycloak server behaves the same way as EAP/WF. You can deploy an app on top of that (as the matter of fact we do this in our quickstarts/examples).

File: secure-embedded-cache/keycloak-setup/install-demo-realm.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#Create roles
+kcadm.sh create roles -r demo -f ${SCRIPT_DIR}/role-admin.json
+kcadm.sh create roles -r demo -f ${SCRIPT_DIR}/role-reader.json
+
+#Create users
+kcadm.sh create users -r demo -f ${SCRIPT_DIR}/user-admin.json
+kcadm.sh create users -r demo -f ${SCRIPT_DIR}/user-reader.json
+
+#Assign roles
+kcadm.sh add-roles -r demo --uusername admin --rolename admin
```
Comment (slaskawi) [inline secure-embedded-cache/keycloak-setup/install-demo-realm.sh:22]: Actually, you can embed users into an exported realm. Here's an example: https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/base/src/test/resources/testrealm.json#L22

File: secure-embedded-cache/pom.xml [hunk 3]
```diff
@@ -65,9 +65,25 @@
...
+         <version>3.0.24.Final</version>
+         <scope>test</scope>
+      </dependency>
+      <dependency>
+         <groupId>org.jboss.resteasy</groupId>
+         <artifactId>resteasy-jaxrs</artifactId>
+         <version>3.0.24.Final</version>
          <scope>provided</scope>
+         <exclusions>
+            <exclusion>
```
Comment (slaskawi) [inline secure-embedded-cache/pom.xml:78]: How about using JEE API jar, like this one: https://mvnrepository.com/artifact/org.jboss.spec/jboss-javaee-all-7.0 ?

**Review by slaskawi — COMMENTED** [review]
Great work! Thanks Martin!

**Comment by mgencur** [comment]
@slaskawi , thanks for the tips. I'll look at them this week.

File: secure-embedded-cache/README.md (hunk 1 above)
Comment (mgencur) [reply secure-embedded-cache/README.md:52]: hi @slaskawi , I was trying to find an example quickstart which is deployed to Keycloak and not WildFly but they seemed to be designed to use both Keycloak and WildFly (this is described in the top-level readme: https://github.com/keycloak/keycloak-quickstarts). I tried deploying to Keycloak directly but I got e.g. this one: https://gist.github.com/mgencur/9c42fbd84b0a7e66d647dca153a3847e
Do you have any example where only Keycloak is used?
Perhaps I'd avoid Docker for this quickstart.

File: secure-embedded-cache/keycloak-setup/install-demo-realm.sh (hunk 2 above)
Comment (mgencur) [reply secure-embedded-cache/keycloak-setup/install-demo-realm.sh:22]: Yeah. That's true. But I wanted to approach this from the end user who needs to create roles and users. Exporting a realm and then adding roles/users to it seems like we're not starting from scratch. Also, when I export the realm it is so long and there are so many elements that 
a) it's confusing for the end user (IMO)
b) personally I don't know which elements I can remove to make it concise and still work

File: secure-embedded-cache/pom.xml (hunk 3 above)
Comment (mgencur) [reply secure-embedded-cache/pom.xml:78]: Looking at the artifact's pom, it has only dependencies on jboss-servlet-api_3.1_spec, jboss-jsp-api_2.3_spec and it shades them inside. Which dependencies you think this new dep should replace? You put this comment on the line where I actually exclude deps. Thanks:)

**Comment by karesti** [comment]
@mgencur hi Martin, do you need any additional review to merge this PR ?

**Comment by mgencur** [comment]
hi @karesti , no, I think this is ready to be integrated. IMO, it's not a good idea to deploy the application to Keycloak directly. It should be a separate entity. Using docker seems unnecessary too.

File: secure-embedded-cache/pom.xml (hunk 3 above)
Comment (slaskawi) [reply secure-embedded-cache/pom.xml:78]: oooh crap! My bad!

File: secure-embedded-cache/keycloak-setup/install-demo-realm.sh (hunk 2 above)
Comment (slaskawi) [reply secure-embedded-cache/keycloak-setup/install-demo-realm.sh:22]: Fair point. I'll leave it up to you.

However, if you'd like to create the smallest json file for import, this should do the trick: https://github.com/stianst/keycloak-demo/blob/master/demo-realm.json

File: secure-embedded-cache/README.md (hunk 1 above)
Comment (slaskawi) [reply secure-embedded-cache/README.md:52]: Yeah, I believe all examples are deployed only on Keycloak: https://github.com/keycloak/keycloak/tree/master/examples

Comment (mgencur) [reply secure-embedded-cache/README.md:52]: OK, thanks. And (AFAIR) all the quickstarts use WildFly and a separate Keycloak server. I'm not sure what is more common, obviously both options are possible.

File: secure-embedded-cache/keycloak-setup/install-demo-realm.sh (hunk 2 above)
Comment (mgencur) [reply secure-embedded-cache/keycloak-setup/install-demo-realm.sh:22]: Thanks for the tip.

**Pull request merged**